server/app/services/doctor_service.py

```diff
@@ -75,3 +79,58 @@ async def set_location_for_doctor(self, doctor_email: str, address: Location):
             return 200, "Address updated successfully"
         except Exception as e:
             return 500, e
+
+
+    async def get_doctor_by_email(self, email: str):
+        """
+        Fetch a doctor's details by email.
+
+        Args:
+            email (str): The email of the doctor.
+
+        Returns:
+            dict: A dictionary containing the doctor's details or None if not found.
+        """
+        try:
+            doctor = await self.doctor_collection.find_one({"email": {"$eq": email}})
+            if doctor:
+                return {
```

Comment:
see how other services are returning

Reply:
doctor object will have many parameters. we need to return only the ones we want to show on the page.